### PR TITLE
goproxy v2

### PR DESCRIPTION
**Pull request by srikrsna-buf**
Docs for the new goproxy

### CODE REVIEW

File: docs/bsr/generated-assets/go.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+This feature is especially useful for creating API clients for Go, as it provides consumers of your
+API with generated Go SDKs on demand. With generated assets, you no longer need to generate Go code
+from Protobuf locally, and thus no longer need to maintain Protobuf files or protoc-based plugins.
+
+## BSR Go module proxy {#proxy}
+
+The BSR Go module proxy implements the [GOPROXY protocol][goproxy] for [Buf modules][modules] by
+generating assets on the fly&mdash;Go code stubs aren't generated until you request them using
+`go get`.
```
Comment (mfridman) [inline docs/bsr/generated-assets/go.md:19]: Should we mention these assets are cached after the first request?

**Review by mfridman — COMMENTED** [review]
Do you think we should have section on the import path for the generated code. I.e., how does one end up with:

```
import buf.build/gen/go/bufbuild/eliza/library/connect-go/buf/connect/demo/eliza/v1/elizav1connect

import elizav1 "buf.build/gen/go/bufbuild/eliza/library/go/buf/connect/demo/eliza/v1"
```

Why is connect suffixed to the end, why is the base types not `.../eliza/v1/elizav1`

File: docs/bsr/generated-assets/go.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    fmt.Println(resp.Msg.Sentence)
+}
+```
+
+Run the module tidy command:
+
+```terminal
+$ go mod tidy
+---
+go: downloading buf.build/gen/go/bufbuild/eliza/library/go v1.28.1-20220706172350-dbde79169a01.7
```
Comment (mfridman) [inline docs/bsr/generated-assets/go.md:144]: There is no mention of versioning, I presume this was on purpose. 

I think versioning should be specific to the implementation, i.e., this Go proxy section should have a section on versioning, likewise the new NPM registry already has this.

File: docs/bsr/generated-assets/go.md (hunk 1 above)
Comment (srikrsna-buf) [reply docs/bsr/generated-assets/go.md:19]: Yes, added that here.

**Review by mfridman — APPROVED** [review]

**Pull request merged**